logger/logger.go

```diff
@@ -1,3 +1,121 @@
 package logger
 
-// TODO: add slog
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"log/slog"
+	"strconv"
+	"sync"
+)
+
+var Logger = slog.New(newHandler(nil))
```

Comment:
Please see the issue #14.

Reply:
Please change the log format to `json` as default.

Reply:
It means remove color and other tings?
```go
L := slog.New(slog.NewJsonHandler(os.Stdout, nil);
```
in this case all is remain is just this line

Reply:
```go
var L = slog.New(newLoggerHandler(nil))

type handler struct {
	h slog.Handler
}

func newLoggerHandler(opts *slog.HandlerOptions) *handler {
	if opts == nil {
		opts = &slog.HandlerOptions{}
	}

	return &handler{
		h: slog.NewJSONHandler(os.Stdout, &slog.HandlerOptions{
			Level:     opts.Level,
			AddSource: true,
		}),
	}
}

func (h *handler) Enabled(ctx context.Context, level slog.Level) bool {
	return h.h.Enabled(ctx, level)
}

func (h *handler) WithAttrs(attrs []slog.Attr) slog.Handler {
	return &handler{h: h.h.WithAttrs(attrs)}
}

func (h *handler) WithGroup(name string) slog.Handler {
	return &handler{h: h.h.WithGroup(name)}
}

func (h *handler) Handle(ctx context.Context, r slog.Record) error {
	return h.h.Handle(ctx, r)
}
```

this is result after of removing previous code.
I still implement the `slog.Handle` interface for future customization(is it ok?)

Reply:
For zap we are able to add multiple core(output streams), if this is possible for slog as well, add both stdout colorized output as well as file output with json format. If this is not possible just add json format for stdout output.

Reply:
I've been exploring the `logger` package and noticed that when creating custom handlers, we need to define methods such as `Enabled()`, `WithAttrs()`, `WithGroup()`, and `Handle()`. While this approach allows for flexibility, I found that using the default handlers like `slog.NewJSONHandler` or `slog.NewTextHandler` can be more convenient as they come with all the necessary methods.

Additionally, I observed that passing attributes without using `slog.AttrType` (e.g., `slog.String("str", "string")`, `slog.Int("int", 1)`, ...) may impact performance negatively.

Considering these points, I suggest that we encourage the use of default handlers and provide clear documentation on the potential performance implications of not using `slog.AttrType`. This could streamline the usage of the library and help users make informed decisions.

I'd love to hear your thoughts on this and if there are any plans or considerations regarding these aspects. Thank you for your work on the `logger` package!

Reply:
I also prefer to use `slog.AttrType`, this improves the performance as well as **safety**. Safety makes our code less error-prone and is more important for me.